### PR TITLE
[Review] Added Initial Test Infrastrusture with Unit Tests

### PR DESCRIPTION
**Pull request by full-of-foo**
TL;DR: Adds an initial test infrastructure (Jasmine) with some unit tests.
Ticket: https://fieldaware.atlassian.net/browse/BI-111

Tasks: 
 - Update tokenReader module to not require a tokens.csv file by parameterising an environment variable
 - Move all app code inside a fa-dw (src) folder to keep project-level files separate (Dockerfiles, Circle.yml etc)
 - Add Jasmine dep and config
 - Add two unit specs
 - Add docker `test` entrypoint to run tests 
 - Added Circle.yml (https://circleci.com/gh/fieldaware/fa-dw-extract/11)

Reviewers:
 - [x] @marzetas 
 - [x] @alvaroortega

### CODE REVIEW

File: fa-dw/app/lib/tokenReader.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+let _parseBizTokensFromString = (csvString) => {
+  let bizTokens = [];
+  let lines = csvString.split('\n');
+
+  if(lines[lines.length-1] === '') lines.pop();
+  for (let line of lines) {
+    let [id, token] = line.split(',');
+    bizTokens.push(new BusinessToken(id, token));
+  }
```
Comment (marzetas) [inline fa-dw/app/lib/tokenReader.js]: Perhaps you could move the check for empty lines inside the for loop, ignore such lines in a more general way.

**Comment by marzetas** [comment]
:+1: (apart from a small comment)

File: fa-dw/app/lib/tokenReader.js (hunk 1 above)
Comment (full-of-foo) [inline fa-dw/app/lib/tokenReader.js]: Nice. How about a validation on each line too (non-empty string cells)?

Comment (marzetas) [inline fa-dw/app/lib/tokenReader.js]: That's what I meant

**Pull request merged**